package.json

```
@@ -2,10 +2,11 @@
   "name": "zod.discriminatedunion",
   "version": "1.0.1",
   "description": "DiscriminatedUnion schema for Zod",
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "types": "dist/cjs/index.d.ts",
```

I used `yarn patch` to try the exports map, and can confirm that it solves the ESBuild problem.

```suggestion
  "types": "dist/cjs/index.d.ts",
  "exports": {
    "types": {
      "import": "./dist/esm/index.d.ts",
      "default": "./dist/cjs/index.js"
    },
    "import": "./dist/esm/index.js",
    "default": "./dist/cjs/index.js"
  },
```